tests/extension.test.ts

```diff
@@ -14,59 +14,119 @@ See the License for the specific language governing permissions and
-import { StatusBarAlignment } from 'vscode';
+import * as cloudBuild from '../src/cloudbuild';
+import {activate, statusBar, deactivate, refreshTicker} from '../src/extension';
+import {StatusBarAlignment} from 'vscode';
+import {GitRepo} from '../src/git';
 
 describe('extension', () => {
-   let context: any;
+  let context: any;
+  let fetchBuildSpy: any;
+  let getCurrentBranchSpy: any;
 
-   beforeEach(() => {
-      vscode.commands.registerCommand.mockReset();
-      vscode.window.createStatusBarItem.mockReset();
-      vscode.window.createStatusBarItem.mockImplementation((align:StatusBarAlignment, prio:number) => {
-         return {
-            alignment: align,
-            priority: prio,
-            text: '',
-            tooltip: '',
-            show: jest.fn()
-         };
-      });
-      cloudbuild.fetchBuilds.mockReset();
-      context = {
-         subscriptions: {
-            push: jest.fn()
-         }
+  beforeEach(() => {
+    vscode.commands.registerCommand.mockReset();
+    vscode.window.createStatusBarItem.mockReset();
+    vscode.window.createStatusBarItem.mockImplementation((align: StatusBarAlignment, prio: number) => {
+      return {
+        alignment: align,
+        priority: prio,
+        text: '',
+        tooltip: '',
+        show: jest.fn(),
       };
-   });
+    });
+    fetchBuildSpy = jest.spyOn(cloudBuild, 'fetchBuilds').mockResolvedValue([]);
+    getCurrentBranchSpy = jest.spyOn(GitRepo.prototype, 'getCurrentBranch').mockResolvedValue('develop');
+    context = {
+      subscriptions: {
+        push: jest.fn(),
+      },
+    };
+  });
 
-   afterEach(() => {
-      deactivate();
-   });
+  afterEach(() => {
+    deactivate();
+  });
+
+  describe('activate', () => {
+    test('should set the status bar', () => {
+      // When
+      activate(context);
+      // Then
+      const statusItemCalls = vscode.window.createStatusBarItem.mock.calls;
+      expect(statusItemCalls).toHaveLength(1);
+      expect(statusItemCalls[0][0]).toBe(vscode.StatusBarAlignment.Left);
+      expect(statusItemCalls[0][1]).toBeGreaterThanOrEqual(1000);
+      expect(statusBar).toBeDefined();
+    });
 
-   describe('activate', () => {
-      test('should set the status bar', () => {
-         // When
-         activate(context);
-         // Then
-         expect(vscode.window.createStatusBarItem.mock.calls).toHaveLength(1);
-         expect(vscode.window.createStatusBarItem.mock.calls[0][0]).toBe(vscode.StatusBarAlignment.Left);
-         expect(vscode.window.createStatusBarItem.mock.calls[0][1]).toBeGreaterThanOrEqual(1000);
-         expect(statusBar).toBeDefined();
+    test('should periodically fetch the builds with folders in workspace', async () => {
+      jest.useFakeTimers();
+      activate(context);
+      await jest.runOnlyPendingTimers();
+      expect(fetchBuildSpy).toHaveBeenCalledWith('develop');
+      // expect(statusBar && statusBar.tooltip).toBe('No build for the current branch yet.');
```

Comment:
Is this comment useful or can it be removed ? (apply for all the comment in this test)

Reply:
I left it here to point that currently, statusBar cannot be checked in specs. I'll remove them for now, and restore these checks with new PR.